### DISCUSSION TITLE
Selecting Bank With Digital Input to Change Address

### DISCUSSION THREAD

**Discussion by ghost** [tttapa/Control-Surface#955] [Q&A] [answered]
hi there, 

is there a way to, for example, have a potentiometer control "General_Purpose_Controller_1" when the input to a digital pin is low, and control "General_Purpose_Controller_2" when the input to the same digital pin is high?

how would i go about coding this using the bank options already available? or would some other elements need to be added?

or maybe there is an even easier way to do this that i'm not seeing. 

thanks.

**Comment by tttapa** [comment, marked as answer]
This is supported by the `SwitchSelector` class: https://tttapa.github.io/Control-Surface-doc/Doxygen/df/def/classSwitchSelector.html

To change from `General_Purpose_Controller_1` to `General_Purpose_Controller_2`, you can use the `CHANGE_ADDRESS` bank type.

See also: https://tttapa.github.io/Control-Surface-doc/Doxygen/d0/d74/Bank_8ino-example.html, https://tttapa.github.io/Control-Surface-doc/Doxygen/da/dc1/FAQ.html#faq-change-address-runtime
